### PR TITLE
CIRC-928: Remove loan action comment when item is recalled.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/CIRC-928 - Recall request retains override comment from check out

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/requests/scenarios/RecallItemsTests.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.LoanPolicyBuilder;
+import lombok.val;
+
+public class RecallItemsTests extends APITests {
+
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/scenarios/RecallItemsTests.java:19]: Personally I don't think there is a need for this blank line

File: src/test/java/api/support/matchers/JsonObjectMatcher.java [hunk 2]
```diff
@@ -50,6 +52,28 @@ public void describeTo(Description description) {
     };
   }
 
+  public static Matcher<JsonObject> hasNoJsonPath(String jsonPath) {
```
Comment (marcjohnson-kint) [inline src/test/java/api/support/matchers/JsonObjectMatcher.java:55]: Does this check that an JSON object does not have a sub-object at that path?

File: src/test/java/api/support/matchers/LoanHistoryMatcher.java [hunk 3]
```diff
@@ -39,9 +41,21 @@ public void describeTo(Description description) {
...
-  public static Matcher<IndividualResource> hasLoanHistory(
-    Matcher<Iterable<? extends JsonObject>> matcher) {
+  @Override
+  protected void describeMismatchSafely(IndividualResource item, Description mismatchDescription) {
+    mismatchDescription.appendText("was ").appendText(item.getJson().toString());
+  }
+
+  @SafeVarargs
+  public static Matcher<IndividualResource> hasLoanHistoryInOrder(
+    Matcher<JsonObject> ... matchers) {
```
Comment (marcjohnson-kint) [inline src/test/java/api/support/matchers/LoanHistoryMatcher.java:51]: Could this fit on a single line of less than 100 characters?

**Review by marcjohnson-kint — APPROVED** [review]

File: src/test/java/api/support/matchers/JsonObjectMatcher.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/test/java/api/support/matchers/JsonObjectMatcher.java:55]: Yes, it actually delegates logic to jayway matcher.

Comment (marcjohnson-kint) [reply src/test/java/api/support/matchers/JsonObjectMatcher.java:55]: Ok, thanks. I was considering giving it a more explicit name, however I think keeping it the same as the jayway matcher is a good idea

**Pull request merged**